R/summary_forest_plot.R

```diff
@@ -0,0 +1,637 @@
+#' @param p.only Number of treatments to plot
+#' @param mtorg Sorting order matrix
+#'
+#' @return Matrix reduction vector
+redu.matrix <- function(ntx, po, p.only, mtorg) {
+  
+  #Correctly create corresponding treatment code no.
+  txcode <- c(1:ntx)
+  #Sorted tx code list
+  st.txcode <- txcode[po[1:p.only]]  #gives same results as po but doing this to be cautious
+  
+  r.mtorg <- mtorg[order(mtorg[, 1]), ]
+  rmt <- (match(r.mtorg[, 2], st.txcode)) + (match(r.mtorg[, 3], st.txcode))
+  
+  return(rmt)
+}
+
+#' Function to reduce the matrix size based on user defined plotting range reduction.
+#'
+#' @param ma Meta-analysis data for direct evidence only
+#' @param rmt Matrix reduction vector
+#'
+#' @return Reduced meta-analysis data
+ma.redu <- function(ma, rmt) {
+  new.lor <- ma$lor[!is.na(rmt), ]
+  new.or <-  ma$or[!is.na(rmt), ]
+  new.predint <-  ma$predint[!is.na(rmt), ]
+  
+  newma <- list(lor = new.lor, or = new.or, predint = new.predint, type = ma$type)
+  return(newma)
+}
+
+#' Function to reduce the matrix size based on user defined plotting range reduction.
+#'
+#' @param mtc Meta-analysis data for direct and indirect evidence
+#' @param rmt Matrix reduction vector
+#' @param p.only Number of treatments to plot
+#' @param po Matrix containing ranking order of treatments
+#'
+#' @return Reduced meta-analysis data
+mtc.redu <- function(mtc, rmt, p.only, po) {
+  #~VECTORS~ inputed mtc === st.mtc # already sorted, just truncate directly
+  new.rank <- mtc$rank[1:p.only, ]
+  if (exists("sucra", where = mtc)) {
+    new.sucra <- mtc$sucra[1:p.only]
+  }
+  
+  #~MATRIX~
+  new.lor <- mtc$lor[!is.na(rmt), ]
+  new.or <-  mtc$or[!is.na(rmt), ]
+  new.predint <-  mtc$predint[!is.na(rmt), ]
+  new.rkgram <- mtc$rkgram[1:(ntx * p.only), ]
+  
+  newmtc <- list(lor = new.lor, or = new.or, predint = new.predint, rkgram = new.rkgram, rank = new.rank, sucra = new.sucra, tau = mtc$tau, type = mtc$type)
+  return(newmtc)
+}
+
+#' Create summary forest matrix plot.
+#'
+#' @param stytitle Title of plot
+#' @param ntx Number of treatments
+#' @param lstx Vector of treatment names
+#' @param mtc Meta-analysis data for direct and indirect evidence
+#' @param ma Meta-analysis data for direct evidence only
+#' @param bpredd TRUE if confidence interval to be plotted as error bars
+#' @param bkey TRUE if key should be included in plot
+#' @param p.only Number of treatments to plot
+#' @param ucex Font size multiplier. Defaults to 1
+mtcMatrixCont <- function(stytitle, ntx, lstx, mtc, ma, bpredd = TRUE, bkey = TRUE, p.only = ntx, ucex = 1) {
+  if (p.only < 3) {
+    stop("Print selection must not be less than 3")
+  } else if (p.only > ntx) {
+    stop("Print selection cannot be more than the total number of interventions")
+  }
+  
+  if (!bkey) {
+    plt.adj <- 0
+  } else if (p.only < ntx) {
+    plt.adj <- 2
+  } else {
+    plt.adj <- 1
+  }
+  
+  sp.order <- "Interventions are displayed sorted by median rank."
```

Comment:
Is this true - or is it being sorted by SUCRA?

I can see in the code that the ranks presented are from `mtc$rank[i %% ntx, 3]`, what value is this? 

Comment also relates to line 630

Reply:
The ranking information is still inconsistent. On the plot, the ranks are given as integers (1, 2, 3, then 4), but if it is SUCRA, then the value would be between 0 and 1 (or 0-100%). I think we may need to take a closer look at `mtc$rank[i %% ntx, 3]`